unstructured/partition/ocr.py

```diff
@@ -315,7 +315,8 @@ def init_table_agent():
 
     global table_agent
 
-    if table_agent is None:
+    if not hasattr(table_agent, "model"):
+        logger.info("Loading the Table Transformer Model...")
```

Comment:
maybe the problem is usage of global here? Not sure why this is necessarily safer. 
we could convert table agent into a singleton class https://python-patterns.guide/gang-of-four/singleton/ 
the example is : https://python-patterns.guide/gang-of-four/singleton/#a-more-pythonic-implementation

Reply:
yeah make sense, i think i was trying to make sure the `table_agent` always has attribute `model` here and i am not sure if a singleton class can help?